### PR TITLE
chore(demo): add search as you type

### LINKED ISSUE #99: Search as you type
As a demo user I would like to search as I type in the search bar instead of having to click enter everytime.

### PR DESCRIPTION
**Pull request by jboix**
## Description

Closes #99

## Changes made

- Introduces a "search as you type" with a debouncing function for the search page.
- Selecting a BU now also triggers a search reload automatically.

### CODE REVIEW

File: demo/src/search/search-page.js [hunk 1]
```diff
@@ -71,12 +81,16 @@ class SearchPage {
...
 
-        await this.search(bu, event.target.value);
-      }
+      await this.search(bu, event.target.value);
+    }, 500));
+
+    this.#dropdownEl.addEventListener('change', async () => {
+      const bu = this.#dropdownEl.value;
+
+      await this.search(bu, this.#searchBarEl.value);
```
[the remark this answers is not in the document]
Comment (jboix) [reply demo/src/search/search-page.js]: Indeed, it does improve the user experience. Thank you for the feedback!

File: demo/src/search/search-page.js [hunk 2]
```diff
@@ -10,6 +10,7 @@ import { openPlayerModal } from '../player/player-dialog';
 import ilProvider from '../core/il-provider';
 import SpinnerComponent from '../core/spinner-component';
 import Pillarbox from '../../../src/pillarbox';
+import { debounce } from '../core/promise-utils';
```
[the remark this answers is not in the document]
Comment (jboix) [reply demo/src/search/search-page.js]: Done !

File: demo/src/core/promise-utils.js [hunk 3]
```diff
@@ -0,0 +1,20 @@
...
+  let timeoutId;
+
+  return function (...args) {
+    clearTimeout(timeoutId);
+
+    timeoutId = setTimeout(() => {
+      func(...args);
+    }, delay);
+  };
+};
```
[the remark this answers is not in the document]
Comment (jboix) [reply demo/src/core/promise-utils.js]: Done !

**Review by amtins — APPROVED** [review]
Tremendous!

**Pull request merged**